### ISSUE TITLE
Minor rendering inaccuracy: objects rendered next to road

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#6130]
On very high zoom levels objects are rendered with a slight offset.

Gate rendered next to the road:
![photo_2018-10-03_22-32-42](https://user-images.githubusercontent.com/1040784/46437737-57dfbf00-c75c-11e8-862f-2695ec03640c.jpg)
geo:51.03329,13.775475?z=22
https://osmand.net/go?lat=51.03329&lon=13.775475&z=22

Traffic lights rendered next to the road:
![photo_2018-10-03_22-32-50](https://user-images.githubusercontent.com/1040784/46437741-5adaaf80-c75c-11e8-9c7a-afee01683d9d.jpg)
geo:51.029697,13.771037?z=22
https://osmand.net/go?lat=51.029697&lon=13.771037&z=22

Both the gate and traffic lights are part of the road, i.e. they share the same nodes.

This minor offset is not really visible on usual zoom levels, only with the two largest zoom levels z21 and z22.

**Comment by vshcherb** [comment]
Yes, this would be hard to fix cause points and lines are generated by "slightly" different grid.

**Comment by scaidermern** [comment]
Another rather unpleasant examples. The building entrances are placed at the wrong positions and the ways are slightly distorted.

Left: OsmAnd, right: editor (JOSM)
![foo](https://user-images.githubusercontent.com/1040784/93455124-2ece7f80-f8dc-11ea-826c-0cf875fddf14.jpg)
